### PR TITLE
Add committer instructions for performing coordinated merges with OMR

### PR DESCRIPTION
**Pull request by 0xdaryl**
Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@fjeremic : as you're the only other that has performed a coordinated merge in the past perhaps you could review this please?

@AdamBrousseau FYI

File: doc/GuideForCommitters.md [hunk 1]
```diff
@@ -110,3 +110,75 @@ first add the `depends:doc` label to the OpenJ9 PR, and then ensure the contribu
...
+    an OMR Acceptance build if one were launched.
+    ```
+    Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#{OMR PR number}
+    ```
+    Before proceeding with the coordinated merge, you must ensure that
+    the build is successful and that it was performed relatively
+    recently (generally within a day of a successful build).
+
+2.  Announce your intention to perform a coordinated merge on the OpenJ9
+    [`committers-public`](https://openj9.slack.com/archives/C8PQL5N65)
```
Comment (fjeremic) [inline doc/GuideForCommitters.md:141]: I would put `#committers-public`.

File: doc/GuideForCommitters.md [hunk 2]
```diff
@@ -110,3 +110,75 @@ first add the `depends:doc` label to the OpenJ9 PR, and then ensure the contribu
...
+1.  Launch a pull request build from the OpenJ9 pull request that you
+    wish to merge that performs the same tests that the OMR Acceptance
+    build.  This ensures that the two changes put together would pass
+    an OMR Acceptance build if one were launched.
+    ```
+    Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#{OMR PR number}
+    ```
+    Before proceeding with the coordinated merge, you must ensure that
+    the build is successful and that it was performed relatively
+    recently (generally within a day of a successful build).
```
Comment (fjeremic) [inline doc/GuideForCommitters.md:138]: Ideally we would ensure the `openj9` and `master` branches of eclipse/openj9-omr are even. If they are not we risk promoting untested commits from OMR. Do you think it's worth noting that?

File: doc/GuideForCommitters.md (hunk 1 above)
Comment (0xdaryl) [reply doc/GuideForCommitters.md]: Agreed.

File: doc/GuideForCommitters.md (hunk 2 above)
Comment (0xdaryl) [reply doc/GuideForCommitters.md:138]: Yes, agreed.  I had it in mind at the outset but forgot to write it down.  Fixed in forced push.

**Review by fjeremic — APPROVED** [review]

**Pull request merged**